fastlane/lib/fastlane/actions/install_on_device.rb

```diff
@@ -9,10 +9,11 @@ def self.run(params)
         end
         taxi_cmd = [
           "ios-deploy",
-          params[:extra],
           "--bundle",
           params[:ipa]
         ]
+
+        taxi_cmd.insert(1, params[:extra]) if params[:extra]
```

Comment:
Instead, maybe just leave the code above as is, but call `taxi_cmd.compact!` afterwards, which should remove all `nil` values 👍